src/node.cc

```diff
@@ -2960,59 +3027,74 @@ void SetupProcessObject(Environment* env,
   // create process.env
   Local<ObjectTemplate> process_env_template =
       ObjectTemplate::New(env->isolate());
-  process_env_template->SetNamedPropertyHandler(EnvGetter,
-                                                EnvSetter,
-                                                EnvQuery,
-                                                EnvDeleter,
-                                                EnvEnumerator,
-                                                env->as_external());
+  process_env_template->SetNamedPropertyHandler(
+    EnvGetter,
+    env->is_main_thread() ? EnvSetter : nullptr,
+    EnvQuery,
+    env->is_main_thread() ? EnvDeleter : nullptr,
+    EnvEnumerator,
+    env->as_external());
   Local<Object> process_env = process_env_template->NewInstance();
   process->Set(env->env_string(), process_env);
 
+  READONLY_PROPERTY(process, "tid", Number::New(env->isolate(),
+                                                env->thread_id()));
   READONLY_PROPERTY(process, "pid", Integer::New(env->isolate(), getpid()));
   READONLY_PROPERTY(process, "features", GetFeatures(env));
   process->SetAccessor(env->need_imm_cb_string(),
                        NeedImmediateCallbackGetter,
                        NeedImmediateCallbackSetter,
                        env->as_external());
 
-  // -e, --eval
-  if (eval_string) {
-    READONLY_PROPERTY(process,
-                      "_eval",
-                      String::NewFromUtf8(env->isolate(), eval_string));
-  }
+  if (env->is_main_thread()) {
+    // -e, --eval
+    if (eval_string) {
+      READONLY_PROPERTY(process,
+                        "_eval",
+                        String::NewFromUtf8(env->isolate(), eval_string));
+    }
 
-  // -p, --print
```

Comment:
Here are a few blocks that are marked as "changes" when in actually just some formatting / ordering changed. Probably not a good idea to offer possible merge conflicts because of indentation fixes?!